config/rootfs/debos/scripts/create_initrd_ramdisk.sh

```diff
@@ -20,6 +20,35 @@ index 4ec926c..ca06c0b 100644
  	# Load ubi with the correct MTD partition and return since fstype
 EOF
 
+patch -p1 << 'EOF'
+--- a/usr/share/initramfs-tools/scripts/nfs-premount/wait_ethernet	1970-01-01 02:00:00.000000000 +0200
++++ b/usr/share/initramfs-tools/scripts/nfs-premount/wait_ethernet	2022-11-25 12:38:46.037498198 +0200
+@@ -0,0 +1,21 @@
++#!/bin/sh
++. /scripts/functions
++
++wait_ethernet() {
++  local netdevwait=60
++  echo "Waiting up to ${netdevwait} secs for any ethernet to become available"
++  while [ "$(time_elapsed)" -lt "$netdevwait" ]; do
```

Comment:
Does `time_elapsed()` get implicitly reset when the script is loaded?  Not sure how it's keeping track of the start time.

Reply:
starttime in time_elapsed() is counted from init script start, so it is limit on total delay, which is really convenient to not exceed lava action timeout.
This is where starttime is set:
https://salsa.debian.org/kernel-team/initramfs-tools/-/blob/master/init#L227
then time_elapsed function:
https://salsa.debian.org/kernel-team/initramfs-tools/-/blob/master/scripts/functions#L154